cbmc/proofs/mqtt_state.c

```diff
@@ -380,9 +380,7 @@ bool valid_IotMqttSubscriptionArray( const IotMqttSubscription_t *pSub,
 
 _mqttSubscription_t *allocate_IotMqttSubscriptionListElt( _mqttSubscription_t *pElt )
 {
-  size_t length;
-  // Assumption avoids arithmetic overflow in malloc, rechecked in valid_*
-  __CPROVER_assume( length < CBMC_MAX_OBJECT_SIZE - sizeof( *pElt ) );
+  uint16_t length;
```

Comment:
This is VERY brittle.  Suppose some day an engineer changes the type of length in the data structure.  I don't think this is worth the change.

Reply:
This is actually necessary for this function.  `length`  is used to set `topicFilterLength`, which is `uint16_t`. There is a conversion error when assigning the `size_t`non-deterministic value to an `uint16_t` variable.